ui/test/extensionUserAgent.test.ts

```diff
@@ -0,0 +1,35 @@
+// tslint:disable:no-useless-files
+
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+ // tslint:disable:no-any max-func-body-length
+
+// Depends on vscode - need to figure out how to enable
```

Comment:
You probably just need to call `registerUIExtensionVariables` with a fake extensionContext (I'm guessing that asAbsolutePath is failing [here](https://github.com/Microsoft/vscode-azuretools/blob/master/ui/src/getPackageInfo.ts#L18). I believe you already figured it out, but these tests are just run using mocha, not VS Code's test runner. It's because they're an npm pacakge and not a VS Code extension and I don't think we'll easily be able to get around that.

Reply:
Actually, I'm getting an error that the vscode module can't be found (whenever tries to import 'vscode', which callWithTelemetryAndErrorHandling.ts does).  Haven't tried very hard to figure this out, may have to do some stubbing/etc.